### PR TITLE
DOC Linting

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-elemental/issues/985

Linting + some very minor changes such as rephrasing things.

### CODE REVIEW

File: docs/en/index.md
Comment (GuySartorelli) [inline docs/en/index.md:1]: This file must not be empty. It must at the very least introduce the module and have a `[CHILDREN]` block. See https://github.com/silverstripe/silverstripe-linkfield/blob/4/docs/en/index.md as an example.

File: docs/en/02_searching-blocks.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# Searching Blocks
+# Searching blocks
 
 ## Overview
 
-Content created through SilverStripe can be searched in various ways.
+Content created through Silverstripe can be searched in various ways.
```
Comment (GuySartorelli) [inline docs/en/02_searching-blocks.md]:
```suggestion
Content created through Silverstripe CMS can be searched in various ways.
```

File: docs/en/03_examples/00_elementalarea_with_dataobject.md [hunk 2]
```diff
@@ -1,16 +1,12 @@
-# Using ElementalArea with DataObject
+# Using `ElementalArea` with `DataObject`
```
Comment (GuySartorelli) [inline docs/en/03_examples/00_elementalarea_with_dataobject.md]: I'm not sure if the docs site uses the first heading or the file name in `[CHILDREN]` block cards when there's no title metadata.... To reduce ambiguity please add a metadata title. e.g:

```text
---
title: Using ElementalArea with DataObject
---
```

Optionally add a `summary` and an `icon` in the metadata as well.

Please apply this to all pages.

File: docs/en/03_examples/00_elementalarea_with_dataobject.md [hunk 3]
```diff
@@ -31,21 +27,15 @@ class BlogPostsAdmin extends ModelAdmin
 ```
 
 `$managed_models` - this static variable connects `BlogPostAdmin` with `BlogPost`, which will be created in the following code.\
-When you create a `DataObject` you should define the connection between your `DataObject` and `ElementalArea` by adding `$has_one` relationship.\
+When you create a `DataObject` you should define the connection between your `DataObject` and [`ElementalArea`](api:DNADesign\Elemental\Models\ElementalArea) by adding `$has_one` relationship.\
```
Comment (GuySartorelli) [inline docs/en/03_examples/00_elementalarea_with_dataobject.md]:
```suggestion
When you create a `DataObject` you should define the connection between your `DataObject` and [`ElementalArea`](api:DNADesign\Elemental\Models\ElementalArea) by adding `$has_one` relationship.
```

File: docs/en/03_examples/00_elementalarea_with_dataobject.md [hunk 4]
```diff
@@ -62,11 +52,13 @@ class BlogPost extends DataObject
...
+[`BaseElement::CMSEditLink()`](api:DNADesign\Elemental\Models\BaseElement::CMSEditLink()) relies on a valid CMS link being available in the parent `DataObject` - in this case, `BlogPost`. It is used to navigate directly to the editing section of the particular element.
+
+> [!NOTE]
+> If you have a nested [`GridField`](api:SilverStripe\Forms\GridField\GridField) this method can get more complicated. Similarly, if your class is used in multiple admins, you will have to choose one to be the canonical admin section for the purposes of this method. This example only shows the simplest case, where the `BlogPost` class is used directly in `BlogPostsAdmin`.
 
 ```php
-// ...
+namespace App\Models;
 
 class BlogPost extends DataObject
```
Comment (GuySartorelli) [inline docs/en/03_examples/00_elementalarea_with_dataobject.md]: Eventually we should probably remove this example and link to [Getting an edit link](https://docs.silverstripe.org/en/5/developer_guides/model/managing_records/#getting-an-edit-link) - you can do that now if you like or we can kick it down the road.

File: docs/en/03_examples/00_elementalarea_with_dataobject.md [hunk 5]
```diff
@@ -62,11 +52,13 @@ class BlogPost extends DataObject
  ```
 
 If you are using `ElementalArea` together with `DataObject`, it is important to define the `CMSEditLink()` method in the class.\
```
Comment (GuySartorelli) [inline docs/en/03_examples/00_elementalarea_with_dataobject.md]:
```suggestion
If you are using `ElementalArea` together with `DataObject`, it is important to define the `CMSEditLink()` method in the class.
```

File: docs/en/03_examples/00_elementalarea_with_dataobject.md [hunk 6]
```diff
@@ -31,21 +27,15 @@ class BlogPostsAdmin extends ModelAdmin
 ```
 
 `$managed_models` - this static variable connects `BlogPostAdmin` with `BlogPost`, which will be created in the following code.\
```
Comment (GuySartorelli) [inline docs/en/03_examples/00_elementalarea_with_dataobject.md]:
```suggestion
`$managed_models` - this static variable connects `BlogPostAdmin` with `BlogPost`, which will be created in the following code.
```

Comment (GuySartorelli) [reply docs/en/03_examples/00_elementalarea_with_dataobject.md]: If these are intended to be paragraph breaks, we should just have an empty line inbetween instead.

File: docs/en/userguide/index.md [hunk 7]
```diff
@@ -3,39 +3,42 @@ title: Content blocks
...
+- [History and rolling back versions](history.md)
+- [Reports](reports.md)
 
 ## Content blocks overview
 
-The [Elemental module](https://addons.silverstripe.org/add-ons/dnadesign/silverstripe-elemental) provides the ability for SilverStripe users to break up a web page into smaller modular parts (or ‘blocks’) that allow content like banners, text and media to be independently managed within the CMS. This also allows for more targeted control over individual types of content, as well as individual versioning, more detailed reports and easy customisation for developers over how certain pieces of content are presented.
+The [Elemental module](https://addons.silverstripe.org/add-ons/dnadesign/silverstripe-elemental) provides the ability for Silverstripe users to break up a web page into smaller modular parts (or ‘blocks’) that allow content like banners, text and media to be independently managed within the CMS. This also allows for more targeted control over individual types of content, as well as individual versioning, more detailed reports and easy customisation for developers over how certain pieces of content are presented.
```
Comment (GuySartorelli) [inline docs/en/userguide/index.md]:
```suggestion
The [Elemental module](https://addons.silverstripe.org/add-ons/dnadesign/silverstripe-elemental) provides the ability for Silverstripe CMS users to break up a web page into smaller modular parts (or ‘blocks’) that allow content like banners, text and media to be independently managed within the CMS. This also allows for more targeted control over individual types of content, as well as individual versioning, more detailed reports and easy customisation for developers over how certain pieces of content are presented.
```

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: docs/en/index.md
Comment (emteknetnz) [reply docs/en/index.md:1]: Updated, have moved most the contents of the readme in here and linked to it from the readme

File: docs/en/02_searching-blocks.md (hunk 1 above)
Comment (emteknetnz) [reply docs/en/02_searching-blocks.md]: Updated

File: docs/en/03_examples/00_elementalarea_with_dataobject.md (hunk 2 above)
Comment (emteknetnz) [reply docs/en/03_examples/00_elementalarea_with_dataobject.md]: Updated

File: docs/en/03_examples/00_elementalarea_with_dataobject.md (hunk 3 above)
Comment (emteknetnz) [reply docs/en/03_examples/00_elementalarea_with_dataobject.md]: Updated

File: docs/en/03_examples/00_elementalarea_with_dataobject.md (hunk 4 above)
Comment (emteknetnz) [reply docs/en/03_examples/00_elementalarea_with_dataobject.md]: Do it later

File: docs/en/03_examples/00_elementalarea_with_dataobject.md (hunk 5 above)
Comment (emteknetnz) [reply docs/en/03_examples/00_elementalarea_with_dataobject.md]: Updated

File: docs/en/03_examples/00_elementalarea_with_dataobject.md (hunk 6 above)
Comment (emteknetnz) [reply docs/en/03_examples/00_elementalarea_with_dataobject.md]: Updated

File: docs/en/userguide/index.md (hunk 7 above)
Comment (emteknetnz) [reply docs/en/userguide/index.md]: Updated

File: docs/en/index.md [hunk 8]
```diff
@@ -0,0 +1,341 @@
...
+
+For a more detailed overview of using this module, please see [the User help guides](docs/en/userguide/index.md).
+
+## Requirements
+
+- Silverstripe CMS ^5.0
+- Versioned Admin ^2.0
+- GridFieldExtensions ^4.0
+
+For a CMS 4.3+ compatible version of this module, please see the [4.x release line](https://github.com/silverstripe/silverstripe-elemental/tree/4#readme).
```
Comment (GuySartorelli) [inline docs/en/index.md]:
```suggestion
```
Requirements are in composer.json, no need to state them here - nor to mention CMS 4 in the CMS 5 docs.

File: docs/en/index.md [hunk 9]
```diff
@@ -0,0 +1,341 @@
...
+In your page type layout template use `$ElementalArea` to render the elements to the page (in place of `$Content`).
+
+## Getting more elements
+
+Note that this module comes by default with the base element and a "Content" element. If you need more, take
+a look at some other modules:
+
+## Silverstripe CMS supported content block modules
+
+To learn more about [Silverstripe CMS supported](https://www.silverstripe.org/software/addons/supported-modules-definition/) content block types see, [Creating new blocks](/docs/en/userguide/edit_content.md).
```
Comment (GuySartorelli) [inline docs/en/index.md]: That definition is way outdated. Just link to https://docs.silverstripe.org/en/project_governance/supported_modules/ instead.

Comment (GuySartorelli) [reply docs/en/index.md]: Also the userdoc link will need to link to the userdocs site explicitly - dev docs won't know how to resolve that path.

File: docs/en/index.md [hunk 10]
```diff
@@ -0,0 +1,341 @@
...
+namespace App\Src\PageTypes;
+
+class MyElementPage extends Page
+{
+}
+```
+
+```yml
+# app/_config/elements.yml
+MyElementPage:
```
Comment (GuySartorelli) [inline docs/en/index.md]:
```suggestion
App\Src\PageTypes\MyElementPage:
```

File: docs/en/index.md [hunk 11]
```diff
@@ -0,0 +1,341 @@
...
+For more information on using this task refer to the [content migration documentation](./09_content_migration.md).
+
+### Customize HTML and markup
+
+The basic element area is rendered into the `DNADesign/Elemental/Models/ElementalArea.ss` template. This loops over
+each of the element controller instances. Each controller instance will render `$ElementHolder` which represents
+the element contained within a holder `div`. The wrapper div is the `ElementHolder.ss` template.
+
+To customise the ElementEditor in the CMS you will need to use the Silverstripe CMS JS Injector to apply transformations
+to the necessary React components. [See here](https://docs.silverstripe.org/en/5/developer_guides/customising_the_admin_interface/how_tos/customise_react_components/)
```
Comment (GuySartorelli) [inline docs/en/index.md]:
```suggestion
to the necessary React components. [See here](https://docs.silverstripe.org/en/developer_guides/customising_the_admin_interface/how_tos/customise_react_components/)
```

File: docs/en/index.md [hunk 12]
```diff
@@ -0,0 +1,341 @@
...
+#### In-line editing
+
+Elements can be edited in the CMS using an inline form where all your elements appear together. For elements
+that are more complex (e.g. use custom `FormField` classes) you can disable the in-line edit form by setting `private static $inline_editable = false` in your
+element class. A [`GridFieldDetailForm`](api:SilverStripe\Forms\GridField\GridFieldDetailForm) will be used to edit blocks that are not in-line editable. Alternatively as the CMS element editor is now React driven, in-line editing functionality can be added to by defining your own React components.
+
+> [!NOTE]
+> The default is that all elements are in-line editable
+
+If in-line editing is not disabled, whilst not having a custom component defined, custom fields will not be rendered unless the field's `schemaDataType` is set See [Framework's FormField definition](https://api.silverstripe.org/5/SilverStripe/Forms/FormField.html#method_getSchemaDataType).
```
Comment (GuySartorelli) [inline docs/en/index.md]: Use the `api:FQCN` link format

File: docs/en/index.md [hunk 13]
```diff
@@ -0,0 +1,341 @@
...
+    private static $controller_template = 'MyElementHolder';
+}
+```
+
+To customise existing block templates such as `Content` and `Form` templates, copy the relevant files from
+`vendor/dnadesign/silverstripe-elemental/templates` to your theme. When doing this, ensure you match the folder
+structure (PHP class namespace) to ensure that your new template version takes priority.
+
+> [!NOTE]
+> The default set of elements follow the [BEM (Block Element Modifier])[(https://getbem.com/](https://getbem.com/) class naming
```
Comment (GuySartorelli) [inline docs/en/index.md]:
```suggestion
> The default set of elements follow the [BEM (Block Element Modifier)](https://getbem.com/) class naming
```

File: docs/en/index.md [hunk 14]
```diff
@@ -0,0 +1,341 @@
...
+```bash
+yarn build
+```
+
+You'll need to have [yarn installed](https://yarnpkg.com/en/docs/install)
+globally in your command line.
+
+> [!NOTE]
+> If adding or modifying colours, spacing, font sizes etc. please try
+> and use an appropriate variable from the silverstripe/admin module if available.
```
Comment (GuySartorelli) [inline docs/en/index.md]:
```suggestion
```
This is all really generic, we don't need it here.

File: docs/en/index.md [hunk 15]
```diff
@@ -0,0 +1,341 @@
...
+
+## Integration with other modules
+
+- [Multiple languages with tractorcow/silverstripe-fluent](docs/en/advanced_setup.md)
```
Comment (GuySartorelli) [inline docs/en/index.md]:
```suggestion
```
This is linked to in the `[CHILDREN]` block below.

File: docs/en/index.md
Comment (GuySartorelli) [inline docs/en/index.md:1]: Most of the docs here do not belong on an index page - please move them into appropriate places in other files.
The index should really just be what the module is, and links to the children pages.

File: docs/en/03_examples/00_elementalarea_with_dataobject.md (hunk 3 above)
Comment (GuySartorelli) [reply docs/en/03_examples/00_elementalarea_with_dataobject.md]: This is still here?

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: docs/en/03_examples/00_elementalarea_with_dataobject.md (hunk 3 above)
Comment (emteknetnz) [reply docs/en/03_examples/00_elementalarea_with_dataobject.md]: Removed

File: docs/en/index.md
Comment (emteknetnz) [reply docs/en/index.md:1]: Updated

File: docs/en/index.md (hunk 8 above)
Comment (emteknetnz) [reply docs/en/index.md]: Updated

File: docs/en/index.md (hunk 9 above)
Comment (emteknetnz) [reply docs/en/index.md:53]: Updated

Comment (emteknetnz) [reply docs/en/index.md]: Updated

File: docs/en/index.md (hunk 10 above)
Comment (emteknetnz) [reply docs/en/index.md]: Updated

File: docs/en/index.md (hunk 11 above)
Comment (emteknetnz) [reply docs/en/index.md]: Updated

File: docs/en/index.md (hunk 12 above)
Comment (emteknetnz) [reply docs/en/index.md]: Updated

File: docs/en/index.md (hunk 13 above)
Comment (emteknetnz) [reply docs/en/index.md]: Updated

File: docs/en/index.md (hunk 14 above)
Comment (emteknetnz) [reply docs/en/index.md]: Removed

File: docs/en/index.md (hunk 15 above)
Comment (emteknetnz) [reply docs/en/index.md]: Removed

File: docs/en/04_defining-you-own-elements.md [hunk 16]
```diff
@@ -0,0 +1,156 @@
...
+## In-line editing
+
+Elements can be edited in the CMS using an inline form where all your elements appear together. For elements
+that are more complex (e.g. use custom `FormField` classes) you can disable the in-line edit form by setting `private static $inline_editable = false` in your
+element class. A [`GridFieldDetailForm`](api:SilverStripe\Forms\GridField\GridFieldDetailForm) will be used to edit blocks that are not in-line editable. Alternatively as the CMS element editor is now React driven, in-line editing functionality can be added to by defining your own React components.
+
+> [!NOTE]
+> The default is that all elements are in-line editable
+
+If in-line editing is not disabled, whilst not having a custom component defined, custom fields will not be rendered unless the field's `schemaDataType` is set with [`FormField::getSchemaDataType()`](api:SilverStripe\Forms\FormField->getSchemaDataType()).
```
Comment (GuySartorelli) [inline docs/en/04_defining-you-own-elements.md]:
```suggestion
If in-line editing is not disabled, whilst not having a custom component defined, custom fields will not be rendered unless the field's `schemaDataType` is set with [`FormField::getSchemaDataType()`](api:SilverStripe\Forms\FormField::getSchemaDataType()).
```
`->` is for properties

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Just one tiny change

File: docs/en/04_defining-you-own-elements.md (hunk 16 above)
Comment (emteknetnz) [reply docs/en/04_defining-you-own-elements.md]: Updated

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**